### DISCUSSION TITLE
Use case: Calculation API

### DISCUSSION THREAD

**Discussion by rasmusfiskerbang** [TanStack/query#1562] [Q&A] [answered]
Hi,

I have a use case where I will ask an API to start a calculation by sending a POST to an endpoint:
`endpoint/calc`
This endpoint will then return a job-id, e.g. `fsdf78dsf9dsg89sfh89`, which will then be used in this endpoint:
`endpoint/result/{job-id}`
This endpoint will return an object with a status which can either be `"success"` or `"pending".` If it is `"pending"` then I want to try again after `retryDelay`.

Any ideas about how I should go about this?

- Should I make the second query dependent on the first?
- What about raising an error if the status is pending, and then pass a function to `retry` which will check whether the error was due to the calculation being pending?

Thanks

**Comment by TkDodo** [comment, marked as answer]
My first thoughts are:

- yes, the second query should be dependent on the existence of an ID in the result of the first query. Use the `enabled` option for that. 
- sounds also good to me! Using the retry function seems nice, but your queryFn needs to throw an error for that. Your query will be in `loading` state until you get a valid result, so you can’t differentiate between loading and pending with that approach I think. If you want that, i think moving the success/pending state to a local useState (write to the state in onSuccess) and then enable `refetchInterval` depending on it would work.

**Reply by rasmusfiskerbang** [reply]
Thanks, glad to hear I was in the right direction.
